### ISSUE TITLE
Output file size with -s or -g

### ISSUE THREAD

**Issue by rg3** [rg3/youtube-dl#6]
Was: http://bitbucket.org/rg3/youtube-dl/issue/141/

If the file size was outputted it would be possible to script youtube-dl to test if the current video the the harddrive is in the best possible quality.

It happens that Youtube re-encode videos, and with higher resolution, so being able to extract the file size from youtube-dl would be very useful.

File size in bytes would be preferred.

**Comment by Bzzz** [comment]
@phihag Another use case for --get-filesize? Determining the size of many videos in advance, like when downloading a complete YouTube channel. Not as a single-option-there-is-your-total-download-size-solution, but with reasonable parsing effort afterwards. For single videos, you can always start the download and stop immediately. For multiple videos (read: possibly hundreds) that's not a practical way to do it. --skip-download skips the download before actually getting/printing the size. 
So, why would I enjoy this feature? I have a medium speed internet connection, too fast to just let it run overnight (might fill up hard disk to the brim if there are large-ish videos that I'm not aware of), but also too slow to just get stuff on demand without annoying everybody, killing VOIP, etc. So I'd like to know how much total DL size will be, so that I can pull a reasonable amount of videos during the idle time at night - finishing before the first user starts in the morning, not accidentally filling my HDD, not wasting precious download time.

**Comment by BlacklightShining** [comment]
Out of curiosity, is this a lot harder to implement than it sounds? I mean, as far as I know, the progress indicators always contain the file size, so it _seems_ like it should be pretty easy. Meanwhile, this request's been open for _literally_ over five years…

**Comment by Hrxn** [comment]
Well, if you have a progress indicator, then yes, that is pretty obvious.

But there are other cases, like RTMP, HLS, HDS, all that streaming crap ;)

You don't have a single video file, instead there are lots of fragments. You somehow could fetch the sizes of these, I think, but yes, it is a lot harder to implement than a single, direct download ;)

**Comment by BlacklightShining** [comment]
@Hrxn Maybe `--get-size` could print just the sizes that are available (so not the ones for streams, for example), and cause `youtube-dl` to exit unsuccessfully if some of them weren't available?

**Comment by Hrxn** [comment]
Yes, that should be possible, but as someone stated earlier, I'm also not sure about a good use case..

**Comment by sebma** [comment]
Hi, 

1°) I suggest to force the user to add the format along with the `--get-size` so that there might be only one result:
`youtube-dl --get-size -f 18 http://www.youtube.com/watch?v=29S1T7RNHGM`
2°) If YouTube does not provide the filesize information for the specified format, you could still get it from the `Content-Length` by doing a HTTP HEAD request (you don't have to download the file :) ), example :

```
$ python
Python 2.7.8 (default, Jun 18 2015, 18:54:19) 
[GCC 4.9.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib, os
>>> link = "http://localhost:1234/Karambiri/La_puissance_de_la_croix_flv_Pasteur_Mamadou_Karambiri_Par_Gael_de_Jesus.mp4"
>>> meta = urllib.urlopen(link).info()
>>> print "Content-Length:", meta.getheaders("Content-Length")[0]
Content-Length: 266655553
>>> 
```

**Comment by sebma** [comment]
Another example with python `httplib`:

```
$ python
Python 2.7.8 (default, Jun 18 2015, 18:54:19) 
[GCC 4.9.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import httplib
>>> fqdn = "cdimage.ubuntu.com"
>>> conn = httplib.HTTPConnection(fqdn)
>>> conn.request("HEAD", "http://cdimage.ubuntu.com/kubuntu/releases/14.04.4/release/kubuntu-14.04.4-desktop-amd64.iso")
>>> response = conn.getresponse()
>>> print "Content-Length:", response.getheader('content-length')
Content-Length: 1105133568
>>> 
```

**Comment by sebma** [comment]
Even easier with python `requests` :) ([Requests: HTTP for Humans](http://docs.python-requests.org/en/latest/user/quickstart/)):

```
$ python
Python 2.7.8 (default, Jun 18 2015, 18:54:19) 
[GCC 4.9.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> url = "http://cdimage.ubuntu.com/kubuntu/releases/14.04.4/release/kubuntu-14.04.4-desktop-amd64.iso"
>>> r = requests.head(url)
>>> print r.headers['content-length']
1105133568
```

**Comment by YosefBerger** [comment]
If we are still looking for use cases, I often have to send videos over messaging applications (whatsapp, messenger etc.), and it would be useful to know if there is a version that is within the file size limits when using the -F flag.

**Comment by Jefferderp** [comment]
I would love a "--size" flag for estimating the final size of a download. Ideally this would act as a dry run, returning the approximate size of a download.

**Comment by blackdimund** [comment]
AtomicParsley 0.9.0 in FreeBSD apparently does not embed thumbnails in video sizes over 4GB. On my arch linux it has AtomicParsley 0.9.6 and it works fine, but FreeBSD hasn't updated AtomicParsley yet. Youtube doesn't give file sizes for a lot of formats like mp4_dash or webm_dash containers. If youtube-dl had a way to get file size it would help avoid embedding files over 4GB.
